src/MSBuild/XMake.cs

```diff
@@ -885,10 +899,24 @@ private static void DebuggerLaunchCheck()
 
                     string timerOutputFilename = Environment.GetEnvironmentVariable("MSBUILDTIMEROUTPUTS");
 
-                    if (outputPropertiesItemsOrTargetResults && targets?.Length > 0 && result is not null)
+                    TextWriter outputStream = null;
+                    try
+                    {
+                        if (outputPropertiesItemsOrTargetResults && targets?.Length > 0 && result is not null)
+                        {
+#pragma warning disable CA2000 // Dispose objects before losing scope is suppressed because the StreamWriter is disposed in the finally block
```

Comment:
If you assigned `new StreamWriter(getResultOutputFile)` into a separate local variable and run `?.Dispose()` on it in the finally block, would it take care of the violation?